bson/src/main/org/bson/json/JsonWriter.java

```diff
@@ -132,31 +132,7 @@ protected void doWriteDateTime(final long value) {
 
     @Override
     protected void doWriteDBPointer(final BsonDbPointer value) {
-        if (settings.getOutputMode() == JsonMode.EXTENDED) {
-            new Converter<BsonDbPointer>() {
-                @Override
-                public void convert(final BsonDbPointer value1, final StrictJsonWriter writer) {
-                    writer.writeStartObject();
-                    writer.writeStartObject("$dbPointer");
-                    writer.writeString("$ref", value1.getNamespace());
-                    writer.writeName("$id");
-                    doWriteObjectId(value1.getId());
-                    writer.writeEndObject();
-                    writer.writeEndObject();
-                }
-            }.convert(value, strictJsonWriter);
-        } else {
```

Comment:
It's been awhile, but I think this is the old STRICT mode representation.  Is it gone now?  I think it would be required for backwards compatibility.

Reply:
Not sure, I can't find any Standard definition of STRICT. It would be different to what 3.0 produced - which wouldn't round trip in that case so wouldn't keep a DBPointer, so the ticket would be "Works as Designed". 

It wouldn't create a valid DBRef when used with Document as the collection would be the namespace and not just the collection.

All for the deprecated Bson DBPointer type. :/

Reply:
Do you think we should revert STRICT back to DBRef style and possibly fix DBRef(String, id) to check for a namespace?

Reply:
Given it's DBPointer and STRICT, I'm now inclined to keep the PR as is.  Round-tripping is probably more important than backwards compatibility for this type.